catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/metacard/workspace/WorkspaceTransformerTest.java

```diff
@@ -127,4 +148,90 @@ public void testMapToMetacardRemapValues() {
   }
+
+  @Test
+  public void testAddListActions() throws MalformedURLException {
+    final WorkspaceMetacardImpl workspaceMetacard = new WorkspaceMetacardImpl();
+    workspaceMetacard.setContent(asList("list1Xml", "list2Xml"));
+    final Action action1 =
+        new ActionImpl(
+            "catalog.data.metacard.list.export",
+            "Title1",
+            "Description1",
+            new URL("http://localhost:1"));
+    final Action action2 =
+        new ActionImpl(
+            "catalog.data.metacard.xml", "Title2", "Description2", new URL("http://localhost:2"));
+    final Action action3 =
+        new ActionImpl(
+            "catalog.data.metacard.list.transform",
+            "Title3",
+            "Description3",
+            new URL("http://localhost:3"));
+    doReturn(newArrayList(action1, action2, action3)).when(actionRegistry).list(anyMap());
+
+    final List<Map<String, Object>> workspaceLists = new ArrayList<>();
+
+    final Map<String, Object> list1AsMap = new HashMap<>();
+    list1AsMap.put(LIST_BOOKMARKS, asList("bookmark1", "bookmark2"));
+    workspaceLists.add(list1AsMap);
+
+    final Map<String, Object> list2AsMap = new HashMap<>();
+    list2AsMap.put(LIST_BOOKMARKS, asList("bookmark3", "bookmark4"));
+    workspaceLists.add(list2AsMap);
+
+    final Map<String, Object> workspaceAsMap = new HashMap<>();
+    workspaceAsMap.put(WORKSPACE_LISTS, workspaceLists);
+
+    // Stub convertDateEntries(Entry<String, Object>) to return the entry.
+    doAnswer(invocationOnMock -> invocationOnMock.getArguments()[0])
+        .when(ut)
+        .convertDateEntries(any());
+
+    wt.addListActions(workspaceMetacard, workspaceAsMap);
+
+    final List<Map<String, Object>> list1Actions =
+        (List<Map<String, Object>>) list1AsMap.get("actions");
+    assertThat(list1Actions, containsInAnyOrder(action(action1), action(action3)));
+
+    final List<Map<String, Object>> list2Actions =
+        (List<Map<String, Object>>) list2AsMap.get("actions");
+    assertThat(list2Actions, containsInAnyOrder(action(action1), action(action3)));
+  }
+
+  private static Matcher<Map<String, Object>> action(Action action) {
+    return new ActionMatcher(action);
+  }
+
+  private static class ActionMatcher extends TypeSafeDiagnosingMatcher<Map<String, Object>> {
+    private final Matcher<Map<String, Object>> matcher;
+
+    private ActionMatcher(Action action) {
+      this(action.getId(), action.getTitle(), action.getDescription(), action.getUrl());
+    }
+
+    private ActionMatcher(String id, String title, String description, URL url) {
+      final Matcher<Map<? extends String, ?>> idMatcher = hasEntry("id", id);
+      final Matcher<Map<? extends String, ?>> titleMatcher = hasEntry("title", title);
+      final Matcher<Map<? extends String, ?>> descriptionMatcher =
+          hasEntry("description", description);
+      final Matcher<Map<? extends String, ?>> urlMatcher = hasEntry("url", url.toString());
```

Comment:
:pencil2: These `Matchers` could follow the naming convention of other entries by describing a `true` or `false` condition, e.g., "`hasUrl`" instead of "`urlMatcher`".